Compile on macOS 14.6.1 fails

Compiling on macOS 14.6.1 fails with:

hexwriter.cpp:50:26: error: calling 'GetHexLine' with incomplete return type 'QString'
            outStream << GetHexLine(temp,2,0,SEGMENT_RECORD) << ENDL;
                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./hexwriter.h:12:13: note: 'GetHexLine' declared here
    QString GetHexLine(uchar *data, uchar size, ushort address ,uchar recordtype);
            ^
/opt/homebrew/lib/QtCore.framework/Headers/qcontainerfwd.h:36:7: note: forward declaration of 'QString'
class QString;
      ^
hexwriter.cpp:52:22: error: calling 'GetHexLine' with incomplete return type 'QString'
        outStream << GetHexLine(reinterpret_cast<uchar*>(&data.data()[(segment-1)*0x10000+address]),16,address,DATA_RECORD) << ENDL;
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./hexwriter.h:12:13: note: 'GetHexLine' declared here
    QString GetHexLine(uchar *data, uchar size, ushort address ,uchar recordtype);
            ^
/opt/homebrew/lib/QtCore.framework/Headers/qcontainerfwd.h:36:7: note: forward declaration of 'QString'
class QString;
      ^
hexwriter.cpp:56:17: error: calling 'GetHexLine' with incomplete return type 'QString'
   outStream << GetHexLine(reinterpret_cast<uchar*>(&data.data()[(segment-1)*0x10000+address]), static_cast<uchar>(size/2),address,DATA_RECORD) << ENDL;
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./hexwriter.h:12:13: note: 'GetHexLine' declared here
    QString GetHexLine(uchar *data, uchar size, ushort address ,uchar recordtype);
            ^
/opt/homebrew/lib/QtCore.framework/Headers/qcontainerfwd.h:36:7: note: forward declaration of 'QString'
class QString;
      ^
hexwriter.cpp:59:17: error: calling 'GetHexLine' with incomplete return type 'QString'
   outStream << GetHexLine(reinterpret_cast<uchar*>(&data.data()[(segment-1)*0x10000+address]),static_cast<uchar>(size),address,DATA_RECORD) << ENDL;
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./hexwriter.h:12:13: note: 'GetHexLine' declared here
    QString GetHexLine(uchar *data, uchar size, ushort address ,uchar recordtype);
            ^
/opt/homebrew/lib/QtCore.framework/Headers/qcontainerfwd.h:36:7: note: forward declaration of 'QString'
class QString;
      ^
hexwriter.cpp:60:17: error: calling 'GetHexLine' with incomplete return type 'QString'
   outStream << GetHexLine(nullptr,0,0,EOF_RECORD) << ENDL;
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./hexwriter.h:12:13: note: 'GetHexLine' declared here
    QString GetHexLine(uchar *data, uchar size, ushort address ,uchar recordtype);
            ^
/opt/homebrew/lib/QtCore.framework/Headers/qcontainerfwd.h:36:7: note: forward declaration of 'QString'
class QString;
      ^
hexwriter.cpp:64:20: error: incomplete result type 'QString' in function definition
QString HexWriter::GetHexLine(uchar *data, uchar size, ushort address, uchar recordtype)
                   ^
/opt/homebrew/lib/QtCore.framework/Headers/qcontainerfwd.h:36:7: note: forward declaration of 'QString'
class QString;
      ^
hexwriter.cpp:66:16: error: 'QString' is an incomplete type
    QString s=(QString(":%1%2%3").arg(size & 0xff, 2, 16, QChar('0')).arg(address, 4, 16, QChar('0')).arg(recordtype, 2, 16, QChar('0')).toUpper()).toLocal8Bit();
               ^
/opt/homebrew/lib/QtCore.framework/Headers/qcontainerfwd.h:36:7: note: forward declaration of 'QString'
class QString;
      ^
hexwriter.cpp:66:34: error: member access into incomplete type 'QString'
    QString s=(QString(":%1%2%3").arg(size & 0xff, 2, 16, QChar('0')).arg(address, 4, 16, QChar('0')).arg(recordtype, 2, 16, QChar('0')).toUpper()).toLocal8Bit();
                                 ^
/opt/homebrew/lib/QtCore.framework/Headers/qcontainerfwd.h:36:7: note: forward declaration of 'QString'
class QString;
      ^
hexwriter.cpp:66:13: error: variable has incomplete type 'QString'
    QString s=(QString(":%1%2%3").arg(size & 0xff, 2, 16, QChar('0')).arg(address, 4, 16, QChar('0')).arg(recordtype, 2, 16, QChar('0')).toUpper()).toLocal8Bit();
            ^
/opt/homebrew/lib/QtCore.framework/Headers/qcontainerfwd.h:36:7: note: forward declaration of 'QString'
class QString;
      ^
hexwriter.cpp:75:22: error: 'QString' is an incomplete type
            s.append(QString("%1").arg(data[i], 2, 16, QChar('0')).toUpper());
                     ^
/opt/homebrew/lib/QtCore.framework/Headers/qcontainerfwd.h:36:7: note: forward declaration of 'QString'
class QString;
      ^
hexwriter.cpp:75:35: error: member access into incomplete type 'QString'
            s.append(QString("%1").arg(data[i], 2, 16, QChar('0')).toUpper());
                                  ^
/opt/homebrew/lib/QtCore.framework/Headers/qcontainerfwd.h:36:7: note: forward declaration of 'QString'
class QString;
      ^
hexwriter.cpp:80:14: error: 'QString' is an incomplete type
    s.append(QString("%1").arg(cs, 2, 16, QChar('0')).toUpper());
             ^
/opt/homebrew/lib/QtCore.framework/Headers/qcontainerfwd.h:36:7: note: forward declaration of 'QString'
class QString;
      ^
hexwriter.cpp:80:27: error: member access into incomplete type 'QString'
    s.append(QString("%1").arg(cs, 2, 16, QChar('0')).toUpper());
                          ^
/opt/homebrew/lib/QtCore.framework/Headers/qcontainerfwd.h:36:7: note: forward declaration of 'QString'
class QString;
      ^
13 errors generated.
make: *** [hexwriter.o] Error 1

--- a/TL866_Updater/QT/hexwriter.h
+++ b/TL866_Updater/QT/hexwriter.h
@@ -1,5 +1,6 @@
 #ifndef HEXWRITER_H
 #define HEXWRITER_H
+#include <QString>
 #include <QTextStream>
 
 class HexWriter

